### PR TITLE
Added FAQ section to docs

### PR DESCRIPTION
**Pull request by galgalesh**
- Added how to run a reactive charm in debug. Fixes https://github.com/juju-solutions/charms.reactive/issues/118
- Added `HACKING.md` with explanation of how to build docs and browse locally

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+```bash
+make docs
+```
+
+To browse the html documentation locally:
+
+```bash
+make docs
+cd docs/_build/html
+python -m SimpleHTTPServer 8765
```
Comment (johnsca) [inline HACKING.md]: You don't actually need to run a server; just pointing to the static HTML files using a `file://` URL works fine.  I usually use `gnome-open docs/_build/html/index.html`

Comment (galgalesh) [reply HACKING.md]: Thats a lot easier!

`gnome-open` isn't installed by default in 17.10, but `xdg-open` is, so I'll use that.

**Comment by galgalesh** [comment]
Fixed the simple server stuff.

Wouldn't it be better to put further debugging information in the Juju docs themselves? So we don't have to maintain this info in two places?

**Comment by johnsca** [comment]
Yep, it would.

**Pull request merged**